### PR TITLE
ENH: Allow custom function for fbp filter

### PR DESCRIPTION
**Pull request by adler-j**
Minor change that allows us to do e.g. lambda tomography easily.

### CODE REVIEW

File: odl/tomo/analytic/filtered_back_projection.py [hunk 1]
```diff
@@ -52,8 +52,12 @@ def _fbp_filter(norm_freq, filter_type, frequency_scaling):
     ----------
     norm_freq : `array-like`
         Frequencies normalized to lie in the interval [0, 1].
-    filter_type : {'Ram-Lak', 'Shepp-Logan', 'Cosine', 'Hamming', 'Hann'}
+    filter_type : {'Ram-Lak', 'Shepp-Logan', 'Cosine', 'Hamming', 'Hann',
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: Why don't we just use the name `filter`?

File: odl/tomo/analytic/filtered_back_projection.py [hunk 2]
```diff
@@ -52,8 +52,12 @@ def _fbp_filter(norm_freq, filter_type, frequency_scaling):
...
+                   `callable`}
         The type of filter to be used.
+        If a string, uses one of the standard filters with that name.
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: "If a string is given, use ..."

File: odl/tomo/analytic/filtered_back_projection.py [hunk 3]
```diff
@@ -52,8 +52,12 @@ def _fbp_filter(norm_freq, filter_type, frequency_scaling):
...
+        If a callable, it must take values in [0, 1] and return the filter
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: "A callable should take an array of values ..."

File: odl/tomo/analytic/filtered_back_projection.py [hunk 4]
```diff
@@ -52,8 +52,12 @@ def _fbp_filter(norm_freq, filter_type, frequency_scaling):
     ----------
     norm_freq : `array-like`
         Frequencies normalized to lie in the interval [0, 1].
-    filter_type : {'Ram-Lak', 'Shepp-Logan', 'Cosine', 'Hamming', 'Hann'}
+    filter_type : {'Ram-Lak', 'Shepp-Logan', 'Cosine', 'Hamming', 'Hann',
+                   `callable`}
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: No backticks

File: odl/tomo/analytic/filtered_back_projection.py [hunk 5]
```diff
@@ -324,10 +331,12 @@ def fbp_filter_op(ray_trafo, padding=True, filter_type='Ram-Lak',
...
         be corrupted due to the circular convolution used. Using padding makes
         the algorithm slower.
-    filter_type : string, optional
-        The type of filter to be used. The options are, approximate order from
-        most noise senstive to least noise sensitive: 'Ram-Lak', 'Shepp-Logan',
-        'Cosine', 'Hamming' and 'Hann'.
+    filter_type : optional
+        The type of filter to be used. The predefined options are, in
+        approximate order from most noise senstive to least noise sensitive:
+        'Ram-Lak', 'Shepp-Logan', 'Cosine', 'Hamming' and 'Hann'.
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: Make this look nicer by using backticks

File: odl/tomo/analytic/filtered_back_projection.py (hunk 1 above)
Comment (adler-j) [reply odl/tomo/analytic/filtered_back_projection.py]: Don't wanna introduce a breaking change

**Comment by adler-j** [comment]
Merge after CI

File: odl/tomo/analytic/filtered_back_projection.py [hunk 6]
```diff
@@ -324,10 +331,13 @@ def fbp_filter_op(ray_trafo, padding=True, filter_type='Ram-Lak',
...
         the algorithm slower.
-    filter_type : string, optional
-        The type of filter to be used. The options are, approximate order from
-        most noise senstive to least noise sensitive: 'Ram-Lak', 'Shepp-Logan',
-        'Cosine', 'Hamming' and 'Hann'.
+    filter_type : optional
+        The type of filter to be used.
+        The predefined options are, in approximate order from most noise
+        senstive to least noise sensitive:
+        `'Ram-Lak'`, `'Shepp-Logan'`, `'Cosine'`, `'Hamming'` and `'Hann'`.
```
Comment (kohr-h) [inline odl/tomo/analytic/filtered_back_projection.py]: Should be double backticks here. You can also make it one listing enclosed by double backticks.

**Pull request merged**